test/proxy_auth_tests.cpp

```diff
@@ -0,0 +1,98 @@
+#include <gtest/gtest.h>
+
+#include <string>
+
+#include <cpr/cpr.h>
+
+// TODO: This uses private proxy servers and public servers for endpoints. This should be replaced with a source
+// code implementation inside server.cpp
+
+#define HTTP_PROXY "192.168.0.180:3128"
+#define HTTPS_PROXY "192.168.0.180:3128"
+#define PROXY_USER "u$er"
+#define PROXY_PASS "p@ss"
+
+using namespace cpr;
+
+// TODO: These should be fixed after a source code implementation of a proxy
+#if defined(false)
+TEST(ProxyAuthTests, SingleProxyTest) {
+    Url url{"https://www.httpbin.org/get"};
```

Comment:
Please do not use external URLs for tests here.
Since those will fail in case we run those test cases offline.

Please use the `HttpServer` for this.
You can find examples for this inside the `get_tests.cpp` file.